packages/block-library/src/cover/test/edit.js

```diff
@@ -0,0 +1,351 @@
+/**
+ * External dependencies
+ */
+import { screen, fireEvent, act, within } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+/**
+ * Internal dependencies
+ */
+import {
+	initializeEditor,
+	registerAllCoreBlocks,
+	unRegisterAllBlocks,
+	createTestBlock,
+} from 'test/integration/helpers/integration-test-editor';
+
+async function setup( testBlock ) {
+	const block = await createTestBlock( testBlock.name, testBlock.attributes );
+	return initializeEditor( { testBlocks: [ block ] } );
+}
+
+async function createAndSelectCoverBlock() {
+	await userEvent.click(
+		screen.getByRole( 'button', {
+			name: 'Color: Black',
+		} )
+	);
+	await userEvent.click(
+		screen.getByRole( 'button', {
+			name: 'Select Cover',
+		} )
+	);
+}
+
+async function selectCoverBlock() {
+	await userEvent.click( screen.getByLabelText( 'Block: Cover' ) );
+}
```

Comment:
I think you should have a helper called `selectBlock`. It's such a common requirement.

Reply:
Good point, done.

Reply:
Maybe we should just use the API from `@wordpress/data`?

```js
window.wp.data.dispatch( blockEditorStore ).selectBlocks();
```

Reply:
Depends on what we want to achieve. I believe the intent of these tests is to always assert on user interactions. If the user can't select the block then the test isn't useful. Using the programmatic method would circumvent this check. Did you see things differently?

Reply:
In this version of the Editor component I couldn't get `selectBlocks` to work, which I think is because it is using the state in the component to store the block rather than the `blockEditorStore` - but not 100% sure of that, could have just been an async timing issue, but I didn't explore further as I agree with Dave that it is probably a good approach to keep these tests as user interaction focused as possible.

Reply:
I think it depends on the tests. If the goal of the test is to test the flow including selecting the blocks then we should use lower-level APIs like `click()`, but if it's just for setting up the editor and not related to the test then it should be ok to use higher-level APIs like `selectBlocks()`.

Ideally, we should have dedicated tests testing on selecting blocks, including but not limited to using a mouse, keyboard, or any other way. Once that's in place, we can just use a simpler API (`selectBlocks()`) for every other tests.

It's not that we _need_ to use `selectBlocks()` everywhere else though, I agree if it's simple enough like a one-liner `click()` then we can just do that. But if we ever need a helper for a more complicated utility, then we should be able to leverage higher-level well-tested APIs to simplify the implementation and increase the performance.